yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnSparkHadoopUtil.scala

```diff
@@ -244,6 +251,47 @@ object YarnSparkHadoopUtil {
     }
   }
 
+  /**
+   * Escapes a string for inclusion in a command line executed by Yarn. Yarn executes commands
+   * using a batch file (.cmd) Windows escaping is neccessary. The
+   * argument is enclosed in single quotes only if there are spaces inside the string.
+   * Also some key characters are escaped that have a meaning in Windows.
+   *
+   * @param arg A single argument.
+   * @return Argument quoted for execution via Yarn's generated shell script.
+   */
+  def escapeForShellWindows(arg: String): String = {
```

Comment:
@vanzin this is not a mere copy. It does add some functionality, and if I don't reinterate the string, I am left with no option but to copy 90% of the code. I wasn't sure about this earlier, but indeed, using CommandBuilderUtils would not be the answer. Or should we change CommandBuilderUtils to include the ' and the % case? The % case is critical, since cmd expects an environment variable.